webrecorder/browsertrix
# Upload WACZ via UI

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/986. Follows https://github.com/webrecorder/browsertrix-cloud/pull/993 API changes



### Changes

Users can now upload .WACZ archives from the "Archived Data" page.

### Manual testing

1. Log in and click "Archived Data"
2. Click "Upload Archive"
3. Click "Browse Files" and select a file. Verify file name and size is shown
4. Click trash icon in file details. Verify file is removed
5. Search for tags and collections. Verify tag and collection input works as expected
6. Re-select a file, enter a name, and click "Upload File". Verify file upload progress is shown
7. Click outside of the modal. Verify confirmation is shown
8. Click "Yes". Verify that upload is canceled via network tab
9. Repeat steps 2-6. Verify upload shows correct progress values
10. Wait for upload to complete. Verify clicking link in toast notification takes you to the upload detail page

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Upload Archive dialog - initial state | <img width="975" alt="Screen Shot 2023-07-18 at 4 43 27 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/7c6ba714-e224-4dbd-8ef6-4660893d4e97"> |
| Upload Archive dialog - File selected and metadata filled | <img width="974" alt="Screen Shot 2023-07-18 at 4 43 59 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/9b515148-8d81-42c6-8c16-e7b25eef3d91"> |
| Upload Archive dialog - Uploading |  <img width="629" alt="Screen Shot 2023-07-19 at 3 44 15 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/951e31dd-e4c9-45f2-a760-9ca8564d700c"> |
| Upload Archive dialog - Cancel confirmation | <img width="663" alt="Screen Shot 2023-07-18 at 5 12 00 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/b5c90d35-9208-48c5-a4e7-b0b46194bddf"> |
| Archived Data - Success toast notification | <img width="432" alt="Screen Shot 2023-07-19 at 2 44 04 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/e2f006e6-119c-41b7-8cda-b20e791aca32"> |

### Follow-ups
- Retry
- Multi-file upload
